### DISCUSSION TITLE
Is it possible to delete MP3s as I listen to them?

### DISCUSSION THREAD

**Discussion by JakeB-OnGithub** [advplyr/audiobookshelf#3058] [Q&A]
I am subscribed to several long-running podcasts, with hundreds of episodes. My ideal scenario is that as a new episode comes out, ABS downloads it, and then when I listen to it, I can delete it.

The problem I'm facing is that if I were to delete the MP3 file, ABS eventually notices and then re-downloads the file.

I can limit the number of episodes back that this goes, but even if I keep it at 1 recent episode, it will just keep downloading that 1 episode over and over again (right?)

Is there some way to say "Don't download episodes you've already downloaded before"?

I have also tried clicking on "remove episode" from within the UI, and it still re-downloads that episode afterwards.

**Comment by nichwall** [comment]
There is not a way to do this yet, but several open feature requests. Once an episode is deleted from ABS, all references to it are removed from the database, so there will need to be some data model changes to support this.

In the meantime, the simplest method will be to set the number of episodes to keep to a high enough number you don't miss any episodes, and then filter by unlistened episodes.

https://github.com/advplyr/audiobookshelf/issues/615
https://github.com/advplyr/audiobookshelf/issues/1840
